project/Build.scala

```diff
@@ -1024,58 +1024,40 @@ object Build {
   lazy val linkerJS: MultiScalaProject = MultiScalaProject(
       id = "linkerJS", base = file("linker/js")
   ).enablePlugins(
-      MyScalaJSPlugin
+      MyScalaJSPlugin, BuildInfoPlugin
   ).zippedSettings("library")(
       commonLinkerSettings _
   ).settings(
       Test / scalacOptions ++= scalaJSCompilerOption("nowarnGlobalExecutionContext"),
 
-      if (isGeneratingForIDE) {
-        unmanagedSourceDirectories in Compile +=
-          baseDirectory.value.getParentFile.getParentFile / "js/src/main/scala-ide-stubs"
-      } else {
-        sourceGenerators in Compile += Def.task {
-          val dir = (sourceManaged in Compile).value
-          val privateLibProducts = (products in (linkerPrivateLibrary, Compile)).value
+      unmanagedSourceDirectories in Compile ++= {
+        if (isGeneratingForIDE)
+          Seq(baseDirectory.value.getParentFile.getParentFile / "js/src/main/scala-ide-stubs")
+        else
+          Seq()
+      },
 
-          val content = {
-            val namesAndContents = for {
-              f <- (privateLibProducts ** "*.sjsir").get
-            } yield {
-              val bytes = IO.readBytes(f)
-              val base64 = java.util.Base64.getEncoder().encodeToString(bytes)
-              s""""${f.getName}" -> "$base64""""
-            }
+      buildInfoPackage in Compile := "org.scalajs.linker.backend.emitter",
+      buildInfoObject in Compile := "PrivateLibData",
+      buildInfoOptions in Compile += BuildInfoOption.PackagePrivate,
+      buildInfoKeys := {
+        val pathsAndContentsTask = Def.task {
```

Comment:
Same here. Maybe we should make this a named task.

Reply:
Similar argument as above: I'm reluctant to make this a build-scoped task because it is essentially a re-shape of `products in (linkerPrivateLibrary, Compile)` (which is already a named task).

So we'd end up with a task of type `Seq[(String, String)]` which really should be a `Seq[File]` :-/